File: src/main/scala-templates/net/jackadull/jackadocs/JackadocsInfo.scala
package net.jackadull.jackadocs

trait JackadocsInfo {
  def artifactID:String = "${project.artifactId}"
  def groupID:String = "${project.groupId}"
  def version:String = "${project.version}"
}
object JackadocsInfo extends JackadocsInfo

File: src/main/java-templates/net/jackadull/jackadocs/JackadocsInfo.java
package net.jackadull.jackadocs;

import net.jackadull.jackadocs.execution.JackadocsProjectInfo;

public final class JackadocsInfo {
  public static final String ArtifactID = "${project.artifactId}";
  public static final String GroupID = "${project.groupId}";
  public static final String Version = "${project.version}";

  public static final JackadocsProjectInfo ProjectInfo = JackadocsProjectInfo.apply(GroupID, ArtifactID, Version);

  private JackadocsInfo() {
    throw new UnsupportedOperationException();
  }
}
